tests/src/common/test_case/response.rs

```diff
@@ -0,0 +1,127 @@
+use sdk::{
+    cosmwasm_ext::InterChainMsg, cosmwasm_std::Coin as CwCoin, testing::InterChainMsgReceiver,
+};
+
+#[must_use]
+#[derive(Debug)]
+pub(crate) struct ResponseWithInterChainMsgs<'r, T> {
+    receiver: &'r mut InterChainMsgReceiver,
+    result: T,
+}
+
+impl<'r, T> ResponseWithInterChainMsgs<'r, T> {
+    pub(super) fn new(receiver: &'r mut InterChainMsgReceiver, result: T) -> Self {
+        Self { receiver, result }
+    }
+
+    pub fn ignore_result(self) -> ResponseWithInterChainMsgs<'r, ()> {
+        ResponseWithInterChainMsgs {
+            receiver: self.receiver,
+            result: (),
+        }
+    }
+
+    #[must_use]
+    #[track_caller]
+    pub fn unwrap_response(mut self) -> T {
+        self.expect_empty();
+
+        self.result
+    }
+}
```

Comment:
Here we use 'result' and 'response' interchangeably. Let's stick to 'response' as being closer to what we are trying to represent.